File: client/ios/NativeModules/counter.m
//#import "React/RCTBridgeModule.h"
//#import "React/RCTEventEmitter.h"
//
//@interface RCT_EXTERN_MODULE(Counter, RCTEventEmitter)
//@end
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Counter, NSObject)
  RCT_EXTERN_METHOD(add)
@end

File: client/ios/NativeModules/counter.swift
//import Foundation
//import React
//
//@objc(Counter)
//class Counter: RCTEventEmitter {
//  private var count = 0 // この行を追加
//
//  @objc(increment)
//  func increment() {
//    count += 1
//    print("count is \(count)")
//    sendEvent(withName: "onIncrement", body: ["count": count])
//  }
//  @objc override class func moduleName() -> String! {
//          return "Counter"
//      }
//
//  override func supportedEvents() -> [String]! {
//    return ["onIncrement"]
//  }
//  
//  @objc override static func requiresMainQueueSetup() -> Bool {
//    return true
//  }
//}
//
import Foundation

@objc(Counter)
class Counter: NSObject {
  private var count = 0

  @objc
  func add() {
    count += 1
    print("count is \(count)")
  }
}
